### PR TITLE
fix: use the o-share js render method for demos

### PR DESCRIPTION
**Pull request by JakeChampion**
using the render method ensures that the demo markup is up-to-date with how sharing is meant to be implemented

this resolves https://github.com/Financial-Times/origami/issues/557

I've also changed the demos to be different templates instead of using one template and several data files as I found it simpler to understand - I'm happy to revert it back to one template if that is what we prefer

screenshot of a demo of o-share which shows that the labels for the links now have accurate descriptions such as "Share on Twitter (opens in a new window)"

### CODE REVIEW

File: components/o-share/demos/src/custom.mustache [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<div data-o-component="o-share" class="o-share">
+	<ul>
+		<li class="o-share__action o-share__action--labelled">
+			<!-- demo only: forms with a submit button may be used for custom share features -->
```
Comment (chee) [inline components/o-share/demos/src/custom.mustache:4]: what does "demo only: " mean here?

Comment (JakeChampion) [reply components/o-share/demos/src/custom.mustache:4]: not really sure, I kept it from the original template - I'll see if the original commit has more context

File: components/o-share/demos/src/horizontal.mustache [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<div
+	data-o-component="o-share"
+	class="o-share"
+	data-o-share-links="twitter facebook linkedin whatsapp pinterest"
+	data-o-share-url="http://on.ft.com/1mUdgA2"
+	data-o-share-titleExtra="FT.com | Pharmaceuticals"
+	data-o-share-summary="US drugs group vows to maintain big British presence"
+	data-o-share-relatedTwitterAccounts="ftcompanies"
+>
```
Comment (chee) [inline components/o-share/demos/src/horizontal.mustache:9]: style nitpick: i think we put the `>` on the same line as the last attr in most of our other html and jsx templates?

File: components/o-share/origami.json [hunk 3]
```diff
@@ -31,47 +31,33 @@
...
-			"data": "demos/src/data-inverse.json",
+			"template": "demos/src/inverse.mustache",
 			"documentClasses": "demo demo--inverse",
 			"description": "inverse share icons"
 		},
 		{
 			"title": "Vertical",
 			"name": "vertical",
-			"template": "demos/src/main.mustache",
-			"data": "demos/src/data-vertical.json",
```
Comment (chee) [inline components/o-share/origami.json:57]: prefer these without the data +1

File: components/o-share/demos/src/custom.mustache (hunk 1 above)
Comment (JakeChampion) [reply components/o-share/demos/src/custom.mustache:4]: I couldn't find more context in the original commit - shall we remove the comment?

Comment (JakeChampion) [reply components/o-share/demos/src/custom.mustache:4]: I suppose it is trying to say that any html can go into the section?

Comment (notlee) [reply components/o-share/demos/src/custom.mustache:4]: Just that it's for demo purposes, Origami doesn't support the share action seen on ft.com, which brings up a share overlay, and is still an ft.com specific implementation. 

This demo was added as part of our work to help remove css overrides overrides:
https://github.com/Financial-Times/next-article/pull/3796/files

**Pull request merged**